### PR TITLE
feat(dashboard-renderer): entity links in TopN table

### PR DESCRIPTION
**Pull request by filipgutica**
# Summary

https://konghq.atlassian.net/browse/MA-2716

- Add a new `entityLink` prop to the dashboard definition that if provided to a TopN table tile, will cause it to use the #name slot to render a link using `<EntityLink>`  instead of displaying as plain text.
- Allow TopN table `card-content` to scroll if the table content is expanding past its container's boundaries. This is necessary for rendering inside dashboard tiles that have fixed dimensions.
- Add a component test case around TopN table with entity links

image

## Gateway Manger MFE (local branch) - using temp package
image

## khcp-ui pr preview with temp package
image

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

### CODE REVIEW

File: packages/analytics/dashboard-renderer/package.json [hunk 1]
```diff
@@ -79,6 +79,7 @@
   },
   "dependencies": {
     "@kong-ui-public/core": "workspace:^",
+    "@kong-ui-public/entities-shared": "workspace:^",
```
Comment (adorack) [inline packages/analytics/dashboard-renderer/package.json:82]: What does this dependency do?

File: packages/analytics/dashboard-renderer/src/components/DashboardRenderer.cy.ts [hunk 2]
```diff
@@ -60,6 +64,7 @@ describe('<DashboardRenderer />', () => {
       }
     }
 
+    // @ts-ignore
```
Comment (adorack) [inline packages/analytics/dashboard-renderer/src/components/DashboardRenderer.cy.ts:67]: You shouldn't need to `ts-ignore` this -- can we talk about the error?

File: packages/analytics/dashboard-renderer/src/components/TopNTableRenderer.vue [hunk 3]
```diff
@@ -10,15 +10,44 @@
       :description="chartOptions.description"
       :synthetics-data-key="chartOptions.syntheticsDataKey"
       :title="chartOptions.chartTitle || ''"
-    />
+    >
+      <template
+        v-if="props.chartOptions?.entityLink"
```
Comment (adorack) [inline packages/analytics/dashboard-renderer/src/components/TopNTableRenderer.vue:15]: Hm.  You shouldn't need conditional chains on `chartOptions` -- it's a required prop.  Did you run into trouble?

**Review by adorack — CHANGES_REQUESTED** [review]

File: packages/analytics/dashboard-renderer/package.json (hunk 1 above)
Comment (filipgutica) [reply packages/analytics/dashboard-renderer/package.json:82]: Need it to use <EntityLink>

File: packages/analytics/dashboard-renderer/src/components/DashboardRenderer.cy.ts (hunk 2 above)
Comment (filipgutica) [reply packages/analytics/dashboard-renderer/src/components/DashboardRenderer.cy.ts:67]:
```
Type 'Promise<{ analytics: true; percentiles: true; api_requests_retention: "24h"; api_requests_retention_ms: number; api_analytics_retention: "30d"; api_analytics_retention_ms: number; }>' is not assignable to type 'Promise<AnalyticsConfig>'.
  Type '{ analytics: true; percentiles: true; api_requests_retention: "24h"; api_requests_retention_ms: number; api_analytics_retention: "30d"; api_analytics_retention_ms: number; }' is not assignable to type 'AnalyticsConfig'.
    Type '{ analytics: true; percentiles: true; api_requests_retention: "24h"; api_requests_retention_ms: number; api_analytics_retention: "30d"; api_analytics_retention_ms: number; }' is not assignable to type 'AnalyticsConfigEnabled'.
      Types of property 'api_requests_retention' are incompatible.
        Type '"24h"' is not assignable to type
```

Comment (filipgutica) [reply packages/analytics/dashboard-renderer/src/components/DashboardRenderer.cy.ts]: it looks like 24h is not a valid `api_requests_retention` value

```
export type AnalyticsRetention = '1d' | '30d' | '90d' | '180d' | '365d'
export type RequestsRetention = '1d' | '7d' | '14d' | '30d'
```

File: packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts [hunk 4]
```diff
@@ -3,6 +3,9 @@ import { ChartMetricDisplay } from '@kong-ui-public/analytics-chart'
 import { DEFAULT_TILE_HEIGHT } from '../constants'
 import type { ExploreFilter, ExploreQuery, TimeRangeV4 } from '@kong-ui-public/analytics-utilities'
 
+export const ENTITY_ID_TOKEN = '{entity-id}'
```
[the remark this answers is not in the document]
Comment (filipgutica) [reply packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts]: done

File: packages/analytics/dashboard-renderer/package.json (hunk 1 above)
Comment (adorack) [reply packages/analytics/dashboard-renderer/package.json:82]: `entities-shared` is massive: 
image

Can you please make it a peer dependency, externalize it in `vite.config.ts`, and add it to the peer dependency requirements listed in the README?  Hopefully that lets you bump the `distSizeChecker` limit back down too.

**Review by adorack — CHANGES_REQUESTED** [review]

**Comment by adamdehaven** [comment]
@adorack is 27KB really that large? (entities-shared)

**Comment by adorack** [comment]
Compared to the size of the overall package, yes.  :)

I don't feel that strongly about it, though -- if you'd recommend against externalizing slightly smaller deps like this, I'm OK.

**Comment by adamdehaven** [comment]
image

For entities-shared, most of the package size is due to:
- `@kong/icons` - this could likely be moved to a peerDependency
- `js-yaml` - I'm unsure on the usage but it looks to be required by the `YamlCodeBlock.vue` component

File: packages/analytics/dashboard-renderer/package.json (hunk 1 above)
Comment (filipgutica) [reply packages/analytics/dashboard-renderer/package.json:82]: So what's the verdict, should this be taken care of in this PR or part of a change to entities-shared like @adamdehaven suggested?

**Review by adorack — APPROVED** [review]

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**